File: isee.py
import requests
import os
from bs4 import BeautifulSoup
from urllib.parse import urlparse, urljoin

# Função para verificar se uma URL possui uma das extensões permitidas
def tem_extensao_permitida(url):
    extensoes_permitidas = ['.mp4', '.mov', '.mp3', '.img', '.jpg', '.jpeg', '.png', '.pdf', '.doc']
    return any(url.endswith(extensao) for extensao in extensoes_permitidas)

# Função para baixar um arquivo
def baixar_arquivo(url, diretorio_destino):
    response = requests.get(url, stream=True)
    if response.status_code == 200:
        nome_arquivo = os.path.join(diretorio_destino, os.path.basename(urlparse(url).path))
        with open(nome_arquivo, 'wb') as arquivo:
            for chunk in response.iter_content(1024):
                arquivo.write(chunk)
        print(f"Arquivo baixado: {nome_arquivo}")
    else:
        print(f"Erro ao baixar arquivo: {url}")

# Função para baixar todos os arquivos permitidos do site
def baixar_arquivos_do_site(soup, url_site, diretorio_destino):
    links = soup.find_all('a', href=True)
    for link in links:
        url_link = link.get('href')
        if url_link and tem_extensao_permitida(url_link):
            url_absoluta = urljoin(url_site, url_link)
            baixar_arquivo(url_absoluta, diretorio_destino)

# Solicitar a URL do site como input
url_site = input("Digite a URL do site: ")

# Criar o diretório de destino para salvar os arquivos
diretorio_destino = input("Digite o diretório de destino para salvar os arquivos: ")
os.makedirs(diretorio_destino, exist_ok=True)

# Fazer a requisição HTTP e obter o conteúdo HTML
response = requests.get(url_site)
html = response.content

# Criar o objeto Beautiful Soup
soup = BeautifulSoup(html, 'html.parser')

# Baixar todos os arquivos permitidos do site
baixar_arquivos_do_site(soup, url_site, diretorio_destino)

File: isee.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');
const { JSDOM } = require('jsdom');
const { promisify } = require('util');

// Função para baixar um arquivo
async function baixarArquivo(url, destino) {
  try {
    const response = await axios.get(url, { responseType: 'stream' });
    const caminhoArquivo = path.join(destino, path.basename(url));
    const writer = fs.createWriteStream(caminhoArquivo);
    response.data.pipe(writer);
    await promisify(writer.on).call(writer, 'finish');
    console.log(`Arquivo baixado: ${caminhoArquivo}`);
  } catch (error) {
    console.error(`Erro ao baixar arquivo: ${url}`);
    console.error(error.message);
  }
}

// Função para baixar um vídeo
async function baixarVideo(url, destino) {
  try {
    const caminhoDestino = path.join(destino, 'videos');
    if (!fs.existsSync(caminhoDestino)) {
      fs.mkdirSync(caminhoDestino, { recursive: true });
    }
    const output = path.join(caminhoDestino, '%(title)s.%(ext)s');
    const { exec } = require('child_process');
    const comando = `youtube-dl -o "${output}" ${url}`;
    await promisify(exec)(comando);
    console.log(`Vídeo baixado: ${url}`);
  } catch (error) {
    console.error(`Erro ao baixar vídeo: ${url}`);
    console.error(error.message);
  }
}

// Função para baixar todas as mídias do site
async function baixarMidiasDoSite(urlSite, destino) {
  try {
    const response = await axios.get(urlSite);
    const dom = new JSDOM(response.data);
    const document = dom.window.document;
    const midias = [...document.querySelectorAll('img, video')];
    
    for (const midia of midias) {
      let urlMidia;
      if (midia.tagName === 'IMG') {
        urlMidia = midia.src;
      } else if (midia.tagName === 'VIDEO') {
        urlMidia = midia.currentSrc;
      }
      
      if (urlMidia) {
        const extensao = path.extname(urlMidia).slice(1).toLowerCase();
        const urlAbsoluta = new URL(urlMidia, urlSite).href;
        
        if (['mp4', 'mov', 'mp3', 'img', 'jpg', 'jpeg', 'png', 'pdf', 'doc', 'avi', 'mkv', 'mpeg', 'vid'].includes(extensao)) {
          if (extensao === 'mp4') {
            await baixarVideo(urlAbsoluta, destino);
          } else {
            await baixarArquivo(urlAbsoluta, destino);
          }
        }
      }
    }
  } catch (error) {
    console.error(`Erro ao baixar mídias do site: ${urlSite}`);
    console.error(error.message);
  }
}

// Solicitar a URL do site
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

readline.question('Digite a URL do site: ', async (urlSite) => {
  // Diretório de destino para salvar as mídias
  const destino = path.join(process.env.HOME || process.env.USERPROFILE, 'Downloads');

  // Criar o diretório de destino, se não existir
  if (!fs.existsSync(destino)) {
    fs.mkdirSync(destino, { recursive: true });
  }

  // Baixar todas as mídias do site
  await baixarMidiasDoSite(urlSite, destino);

  // Exibir o caminho onde as mídias foram salvas
  console.log(`Mídias salvas em: ${destino}`);

  readline.close();
});
